docs/contribute/DEV_DOCS.md

```diff
@@ -149,7 +149,15 @@ During development you can run static type checks using [mypy][]:
     $ mypy
     # No output is good!
 
-There are also many [editor integrations for mypy][].
+and [pyright][]:
+
+    $ npx pyright
+    ...
+    Found 40 source files
+    0 errors, 0 warnings, 0 infos
+    Completed in 2sec
+
+There are also many [editor integrations for mypy][], and Pyright is integrated into VS Code's Python support.
```

Comment:
FWIW, to get this working nicely in VS Code I needed:
* Using [pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) (I'm guessing most people will have this already)
* Setting `python.analysis.typeCheckingMode = basic` (default: off)

I think the general view of the group has been to not document any editor-specific stuff, but I'd support a sentence or two about how to get it working in vs-code.